src/protocol/tx-validity.md

```diff
@@ -179,6 +179,12 @@ def metadata_gas_fees(tx) -> int:
     elif tx.type == TransactionType.Script:
         # add intrinsic cost of calculating the transaction id
         total += sha256_gas_fee(size(tx))
+    elif tx.type == TransactionType.Upgrade:
+        # add intrinsic cost of calculating the transaction id
+        total += sha256_gas_fee(size(tx))
```

Comment:
Isn't this same for all tx types? Could we maybe move it outside the if statement, e.g. to the initializer of `total`? Also above we add it for every created output contract which doesn't seem correct. Not really part of this PR, but noticed that while reviewing.